ndcube/ndcube.py

```diff
@@ -918,6 +919,37 @@ def __neg__(self):
         return self._new_instance(data=-self.data)
 
     def __add__(self, value):
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                    value_data = value.data * value.unit.to(self.unit)
```

Comment:
I think to be consistent, you should strip the unit after conversion here by using `.to_value` rather than `.to`
```suggestion
                    value_data = value.data * value.unit.to_value(self.unit)
```